lxd/storage_volumes.go

```diff
@@ -203,44 +205,53 @@ func storagePoolVolumesGet(d *Daemon, r *http.Request) response.Response {
 		return response.SmartError(err)
 	}
 
-	// Get all instance volumes currently attached to the storage pool by ID of the pool and project.
-	volumes, err := d.db.Cluster.GetStoragePoolVolumes(projectName, poolID, supportedVolumeTypesInstances)
-	if err != nil && !response.IsNotFoundError(err) {
-		return response.SmartError(err)
-	}
+	var volumes []*api.StorageVolume
+	if allProjects {
+		// Get all instance volumes currently attached to the storage pool by ID of the pool and project.
+		volumes, err = d.db.Cluster.GetStoragePoolVolumesAllProjects(poolID, supportedVolumeTypes)
+		if err != nil && !response.IsNotFoundError(err) {
+			return response.SmartError(err)
+		}
```

Comment:
@tomponline I noticed there is some special handling of image volumes when getting volumes by project. `GetStorageVolumesAllProjects` will set the project of all image volumes to `default`. Do you think that is an issue?